spec/datadog/appsec/actions_handler/serializable_backtrace_spec.rb

```diff
@@ -0,0 +1,176 @@
+# frozen_string_literal: true
+
+require 'datadog/appsec/spec_helper'
+
+RSpec.describe Datadog::AppSec::ActionsHandler::SerializableBacktrace do
+  let(:location_struct) do
+    Struct.new(:path, :lineno, :label) do
+      def to_s
+        "#{path}:#{lineno}:in `#{label}'"
+      end
+    end
+  end
+
+  describe '#to_msgpack' do
+    before do
+      Datadog.configuration.appsec.stack_trace.max_depth = 40
+      Datadog.configuration.appsec.stack_trace.top_percentage = 75
+    end
+
+    after do
+      Datadog.configuration.appsec.reset!
+    end
+
+    it 'correctly serializes stack attributes' do
+      result = pack_and_unpack(described_class.new(locations: [], stack_id: 'some-id'))
+
+      expect(result.fetch('id')).to eq('some-id')
+      expect(result.fetch('language')).to eq('ruby')
```

Comment:
Just throwing idea (if you like it)

```suggestion
      expect(result).to include('id' => 'some-id', 'language' => 'ruby')
```